src/cdk-experimental/menu/menu-item.ts

```diff
@@ -50,9 +52,52 @@ export class CdkMenuItem {
   }
   private _checked = false;
 
+  /**  Whether the CdkMenuItem is disabled - defaults to false */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
   /** Emits when the attached submenu is opened */
   @Output() opened: EventEmitter<void> = new EventEmitter();
 
+  constructor(
+    /** reference a parent CdkMenuGroup component */
+    private _menuGroup: CdkMenuGroup
+  ) {}
+
+  /** Configure event subscriptions */
+  ngAfterContentInit() {
+    if (this.role !== 'menuitem') {
+      this._menuGroup.change.subscribe((button: CdkMenuItem) => this._toggleCheckedState(button));
```

Comment:
Typically, we'd make sure to unsubscribe any subscriptions in ngOnDestroy, using a ReplaySubject and takeUntil (https://ultimatecourses.com/blog/exploring-angular-lifecycle-hooks-ondestroy). I don't know if there's a different convention for this in the CDK, though.

Reply:
My approach here is that the emitter comes from the parent and the child wouldn't get destroyed  independent of the parent hence it should auto-unsubscribe. I've seen it in a few places in the cdk with that assumption IIRC.

Reply:
That should be fine, although it's worth considering the lazy-loaded scenario, where menu items are generated on demand. If previous menu items are destroyed (such as with a long, scrollable menu list), their subscriptions will have to be unsubscribed manually.

That said, that probably won't be in place for a while, so it should be okay to follow the convention for now.

Reply:
It's actually pretty common to use `ngIf` or `ngFor` with menu items, which will destroy the items independently from the parent. You can just do the right thing here by using `takeUntil(this._destroyed)` with the subscription (you can see a bunch of examples of this throughout the library)